app/assets/javascripts/notices.js

```diff
@@ -0,0 +1,86 @@
+var Notice = (function() {
```

Comment:
Yes, I have yet to check it in. I'm still working on a few tests for the udpates to the angular controllers.